### ISSUE TITLE
Creating pipes for GWakeup: Too many open files

### ISSUE THREAD

**Issue by jonaski** [strawberrymusicplayer/strawberry#724]
**Describe the bug**
Crashes with "Creating pipes for GWakeup: Too many open files"

**To Reproduce**
Play for 30 minutes or more.

**System Information:**
 - Operating system: openSUSE Tumbleweed and Leap 15.3
 - Strawberry Version: 0.9.3-103-gf786a170

**Comment by jonaski** [comment]
@gavinhoward I'm getting a crash with "Creating pipes for GWakeup: Too many open files" on openSUSE, looks like it's a file descriptor leak somewhere. I've tested with Qt 6 only. Have you seen this?

**Comment by jonaski** [comment]
I can't see that I've changed anything recently that should cause a socket leak.
So maybe it's in Qt 6, I saw this:
https://code.qt.io/cgit/qt/qtbase.git/commit/?h=6.1&id=93450a6ab263bc940ce4fc82836c965a5ea9bb4e

**Comment by gavinhoward** [comment]
I have not seen this, but that's because I don't build with Qt6. I will do a build with that and see. It will take me a few days though.

**Comment by jonaski** [comment]
It crashes with Qt 5 too. It occurs circa after 2-3 hours of playing FLAC's.
It could be a glib issue on openSUSE issue.
I suspect it happened after I upgraded openSUSE from 15.2 to 15.3, or it has to be related to one of the recent changes I've done in strawberry.

**Comment by gavinhoward** [comment]
Hmm...I run my build with Qt5, and I almost exclusively play FLAC's. Yet I have not seen the crash.

Let me do `git pull`, build, and try with that. Maybe that will help narrow the issue to openSUSE or recent changes.

**Comment by jonaski** [comment]
Checking with: `ls -la /proc/$(pidof strawberry)/fd/ | wc -l`

On openSUSE tumbleweed both with Qt 6.1.1 and Qt 5.15.2 it looks like it's increasing file descriptors by 4 on each song change.
On Fedora 33 with Qt 5.15.2 it's pretty steady under 100. I also played for several hours without running out of file descriptors, but it crashed because of the Qt/wayland connection lost bug.

**Comment by gavinhoward** [comment]
Okay. I have been testing my build of recent changes, and I haven't seen a crash. However, I will start running the command-line you gave to check.

**Comment by jonaski** [comment]
I think the problem is coming from commit https://github.com/strawberrymusicplayer/strawberry/commit/67f831bebacd396f1b9f3bbdef8ef94de461570d
The systemtrayicon changes I did. The reason it works on Fedora is because it's Gnome without systemtray.

**Comment by gavinhoward** [comment]
That might be it because I run the awesome window manager, which doesn't have systemtray. Or at least, I don't have it on mine.

**Comment by jonaski** [comment]
It's QSystemTrayIcon::isSystemTrayAvailable() is opening file descriptors and not closing them, it seems like a Qt bug.

**Comment by jonaski** [comment]
I've changed the behavior to only call QSystemTrayIcon::isSystemTrayAvailable() in the constructor and save the result to a variable. I'm not sure it's a good idea to call it on each song change anyway.
Now it's steady on 63 open file descriptors.
I never guessed that it had to do with the system tray icon.
I'll look into the Qt sources tomorrow to see if I can submit a fix to Qt.

**Comment by getlogan** [comment]
Hi @jonaski , I am getting this same issue with files in my home Music directory on an m.2 SSD, (so NOT over network).

I am using Strawberry 1.2.11 flatpak version on Pop!_OS 22.04 LTS.

I can reproduce it when I jump between tracks for long enough.

Using the following command I saw you using in a different issue:
`ls -la /proc/$(pidof strawberry)/fd/ | wc -l`

I can see the number will eventually go up to 1024 and then jumping to another track will crash Strawberry.

Also, this might have something to do with another issue I have noticed. If holding down the F8 key (next track) in Strawberry for even a few seconds, the whole OS will lock up for minutes at a time.
